plugin.rb

```diff
@@ -332,3 +332,43 @@ class ::ListController
     )
   end
 end
+
+Rake::Task.define_task code_review_full_sha_backfill: :environment do
+  posts_with_commit = Post
+    .joins("INNER JOIN topics ON topics.id = posts.topic_id")
+    .joins("INNER JOIN topic_custom_fields ON topics.id = topic_custom_fields.topic_id")
+    .includes(topic: :_custom_fields)
+    .where("topic_custom_fields.name = '#{DiscourseCodeReview::COMMIT_HASH}' AND
+            topics.deleted_at IS NULL AND
+            posts.deleted_at IS NULL AND
+            posts.post_number = 1 AND
+            posts.raw ~ 'sha: [0-9a-f]{6,10}' AND
+            posts.raw !~ 'sha: [0-9a-f]{11,60}'")
```

Comment:
I realise this is a bit weird, but for the life of me I could not get this to match with the end of string `$` delimiter or the start/end of word `\y` like I was able to in ruby with the `\b` below.

It works though, and ensures idempotency by not returning the already fixed posts when querying again.

Reply:
yeah... way to rare to have a commit with the description `raw: abc287abc123` should be safe. 

That said... this works: 

```
DB.query("select 1 where regexp_match('hello\nworld','hello$', 'm') IS NOT NULL")
=> [#<#<Class:0x00005633caa1cc38>:0x00005633c9cf49e8 @column0=1>]
```